### PR TITLE
Upgrade dependency management to OSGi R7

### LINKED ISSUE #182: Upgrade dependency management to OSGi R7
Since AEM 6.5 OSGi R7 is fully supported (and with AEM 6.4 at least partially, compare with https://helpx.adobe.com/experience-manager/kt/eseminars/gems/Using-OSGi-R7-in-AEM.html). The according dependencyManagement in the parent (https://github.com/adobe/aem-project-archetype/blob/4774148f1b2f255c2461313c19c533df60bfb000/src/main/archetype/pom.xml#L492) should be updated to reference the supported versions in AEM 6.5. This most probably requires a new version of the archetype though.
Also instead of the aggregate artifacts the individual ones should be used (to keep the classpath as lean as possible). Both artifact types can be defined in the dependencyManagement, though. For further information about individual vs. aggregate dependencies please refer to https://issues.apache.org/jira/browse/SLING-7384?focusedCommentId=16327433&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16327433

### PR DESCRIPTION
**Pull request by bpauli**
fixes #182

### CODE REVIEW

**Comment by jckautzmann** [comment]
Once the archetype is updated, will it still work with AEM 6.3 and 6.4?

File: src/main/archetype/core/pom.xml [hunk 1]
```diff
@@ -101,15 +101,23 @@
...
+            <artifactId>org.osgi.service.metatype.annotations</artifactId>
         </dependency>
         <dependency>
             <groupId>org.osgi</groupId>
-            <artifactId>osgi.annotation</artifactId>
+            <artifactId>org.osgi.service.component</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>osgi.core</artifactId>
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: Why still the aggregate dependency for core? Please reference framework, config admin, event ... rather via separate artifacts

File: src/main/archetype/core/pom.xml [hunk 2]
```diff
@@ -101,15 +101,23 @@
         <!-- OSGi Dependencies -->
         <dependency>
             <groupId>org.osgi</groupId>
-            <artifactId>osgi.core</artifactId>
+            <artifactId>org.osgi.service.component.annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
```
Comment (kwin) [inline src/main/archetype/core/pom.xml]: This is an aggregate. There is the version and bundle annotations.

**Comment by bpauli** [comment]
@jckautzmann Yes, with the latest commit https://github.com/adobe/aem-project-archetype/pull/207/commits/133858fefcd9b283ddce0aa162a5029e1a39021b the bundle can also be resolved in 6.3.3

**Pull request merged**